data_diff/database.py

```diff
@@ -183,6 +270,11 @@ def md5_to_int(self, s: str) -> str:
     def to_string(self, s: str):
         return f"{s}::varchar"
 
+    def canonize_by_type(self, value, coltype: ColType) -> str:
+        if isinstance(coltype, (Timestamp, TimestampTZ)):
+            return self.to_string(f"{value}::timestamp({coltype.precision})")
```

Comment:
Objective (3) I think is to ensure that we 'downcast' the precision between databases, which is what Chipper also ran into. 

Wouldn't that require us to set the precision to `min(coltype.precision)` to negotiate to the lowest precision between the two different database's timestamp columns? `timestamp(9)` in Snowflake, and `timestamp(6)` in Postgres should be able to checksum to the same number which was that motivating case from Chipper.

Reply:
Yes, it would require that.